src/utils/string/string.test.ts

```diff
@@ -0,0 +1,25 @@
+import { constructJiraString, copyToClipboard } from ".";
+
+describe("copyToClipboard", () => {
+  it("should copy the correct text", () => {
+    Object.defineProperty(navigator, "clipboard", {
+      value: {
+        writeText: jest.fn(),
+      },
+    });
+    copyToClipboard("copy text");
+    expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1);
+    expect(navigator.clipboard.writeText).toHaveBeenCalledWith("copy text");
+  });
+});
+
+describe("constructJiraString", () => {
+  it("should properly format text", () => {
```

Comment:
nit. Separate this out into more test cases. So this can also serve as documentation for how the function is supposed to work.

As an example
"should add an ellipsis between lines if they aren't adjacent to each other"
"should print adjacent lines side by side"
"should properly format text"
"should print nothing when there are no bookmarks"
"should handle out of bounds bookmarks"